### PR TITLE
GAWB-3001: Add Data Use to edit/update consent modal

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadinstitute.atlassian.net/browse/GAWB-3001

## See Also
* Consent PR: https://github.com/DataBiosphere/consent/pull/250
* ORSP PR is dependent on this: broadinstitute/orsp#331
* FC-Develop PR: broadinstitute/firecloud-develop#940

## Testing
As an admin, test the add new consent and edit consent modals.

### CODE REVIEW

File: src/app/modals/dul-modal/add-dul-modal.html [hunk 1]
```diff
@@ -47,11 +47,23 @@ <h2 class="cm-title-admin dul-color">
                         <textarea ng-model="useRestriction"
                                   name="inputSDUL"
                                   class="form-control col-lg-12 vote-input"
-                                  placeholder="Structured string of the Data Use Limitations (JSON format)" required>
+                                  placeholder='Structured string of the Data Use Limitations (JSON format, e.g. {"type":"everything"})' required>
```
Comment (zarsky-broad) [inline src/app/modals/dul-modal/add-dul-modal.html]: Use double quotes to wrap the string, and escape internal quotes with `&quot;`

File: src/app/modals/dul-modal/add-dul-modal.html [hunk 2]
```diff
@@ -47,11 +47,23 @@ <h2 class="cm-title-admin dul-color">
...
                         </div>
                     </div>
+
+                    <div class="form-group admin-form-group ">
+                        <label class="col-lg-3 col-md-3 col-sm-3 col-xs-4 control-label dul-color">Data Use</label>
+                        <div class="col-lg-9 col-md-9 col-sm-9 col-xs-8 admin-input">
+                        <textarea ng-model="dataUse"
+                                  name="inputDU"
+                                  class="form-control col-lg-12 vote-input"
+                                  placeholder='Structured string of the Data Use Questions/Answers (JSON format, e.g. {"generalUse":true})' required>
```
Comment (zarsky-broad) [inline src/app/modals/dul-modal/add-dul-modal.html]: See above

File: src/app/modals/dul-modal/dul-modal.controller.js [hunk 3]
```diff
@@ -65,11 +77,19 @@
             try{
                 JSON.parse($scope.useRestriction);
             }catch(err){
-                $scope.duplicateEntryAlert(0, "Unable to process JSON");
+                $scope.duplicateEntryAlert(0, "Unable to process Structured Limitations JSON");
```
Comment (zarsky-broad) [inline src/app/modals/dul-modal/dul-modal.controller.js]: Can this and the previous identical try/catch block be made into a function that's called to de-dupe the code?

File: src/app/modals/dul-modal/edit-dul-modal.html [hunk 4]
```diff
@@ -45,9 +45,19 @@ <h2 class="cm-title-admin dul-color">
                         <label class="col-lg-3 col-md-3 col-sm-3 col-xs-4 control-label dul-color">Structured Limitations</label>
                         <div class="col-lg-9 col-md-9 col-sm-9 col-xs-8 admin-input">
                             <textarea ng-model="useRestriction"
-                                   name="inputSDUL"
-                                   class="form-control col-lg-12 vote-input"
-                                   placeholder="Structured string of the Data Use Limitations (JSON format)" required>
+                                      name="inputSDUL"
+                                      class="form-control col-lg-12 vote-input"
+                                      placeholder='Structured string of the Data Use Limitations (JSON format, e.g. {"type":"everything"})' required>
```
Comment (zarsky-broad) [inline src/app/modals/dul-modal/edit-dul-modal.html]: See above

File: src/app/modals/dul-modal/edit-dul-modal.html [hunk 5]
```diff
@@ -45,9 +45,19 @@ <h2 class="cm-title-admin dul-color">
...
+                            </textarea>
+                        </div>
+                    </div>
+                    <div class="form-group admin-form-group">
+                        <label class="col-lg-3 col-md-3 col-sm-3 col-xs-4 control-label dul-color">Data Use</label>
+                        <div class="col-lg-9 col-md-9 col-sm-9 col-xs-8 admin-input">
+                            <textarea ng-model="dataUse"
+                                      name="inputSDUL"
+                                      class="form-control col-lg-12 vote-input"
+                                      placeholder='Structured string of the Data Use Questions/Answers (JSON format, e.g. {"generalUse":true})' required>
```
Comment (zarsky-broad) [inline src/app/modals/dul-modal/edit-dul-modal.html]: See above

**Review by zarsky-broad — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -7,20 +7,28 @@ consent-ui is an angularJS application that requires running consent and consent
...
+   From `firecloud-develop`:
+   
+    ```bash
+    RUN_CONTEXT=local APP_NAME=consent-ui ENV=dev OUTPUT_DIR=$PWD/../consent-ui/config ./configure.rb
     ```
-4. Run the docker compose file 
-    ```
-    docker-compose -p consent-ui -f config/docker-compose.yaml up
+
+2. One time commands - only need to install npm once and build once:
```
Comment (davidangb) [inline README.md]: nit: markdown will render this as "3."

**Review by davidangb — APPROVED** [review]
on minor nit inline, otherwise :+1:, did not test

**Comment by rushtong** [comment]
Thank you @zarsky-broad for the `npm` changes - local running is much simpler now! Please have a second look at the de-duping method if you wish to re-review.

**Pull request merged**